packages/fe-fpm-writer/src/action/index.ts

```diff
@@ -0,0 +1,88 @@
+    };
+    setCommonDefaults(config, manifestPath, manifest);
+
+    // set default values for visibility and enabled
+    config.settings.enabled = config.settings.enabled || true;
+    config.settings.visible = config.settings.visible || true;
+
+    return config as InternalCustomAction;
+}
+
+export function getTargetElementReference(manifest: any, target: CustomActionTarget): any {
+    const page = manifest['sap.ui5'].routing.targets[target.page];
+    page.options = page.options || {};
+    page.options.settings = page.options.settings || {};
+    if (target.control === TargetControl.header || target.control === TargetControl.footer) {
+        page.options.settings.content = page.options.settings.content || {};
+        page.options.settings.content[target.control] = page.options.settings.content[target.control] || {};
+        page.options.settings.content[target.control].actions =
+            page.options.settings.content[target.control].actions || {};
+        return page.options.settings.content[target.control].actions;
+    } else {
+        const controlPrefix = target.navProperty ? target.navProperty + '/' : '';
+        const controlSuffix = target.qualifier ? '#' + target.qualifier : '';
+        const controlId = `${controlPrefix}${target.control}${controlSuffix}`;
+        page.options.settings.controlConfiguration = page.options.settings.controlConfiguration || {};
+        page.options.settings.controlConfiguration[controlId] =
+            page.options.settings.controlConfiguration[controlId] || {};
+        page.options.settings.controlConfiguration[controlId].actions =
+            page.options.settings.controlConfiguration[controlId].actions || {};
+        return page.options.settings.controlConfiguration[controlId].actions;
+    }
+}
+
+/**
+ * Add a custom page to an existing UI5 application.
+ *
+ * @param {string} basePath - the base path
+ * @param {CustomAction} actionConfig - the custom action configuration
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns {Promise<Editor>} the updated memfs editor instance
+ */
+export function generateCustomAction(basePath: string, actionConfig: CustomAction, fs?: Editor): Editor {
+    validateVersion(actionConfig.ui5Version);
+    if (!fs) {
+        fs = create(createStorage());
+    }
+
+    const manifestPath = join(basePath, 'webapp/manifest.json');
+    const manifest = fs.readJSON(manifestPath) as Manifest;
+
+    const config = enhanceConfig(actionConfig, manifestPath, manifest);
+
+    const root = join(__dirname, '../../templates/action');
+
+    // enhance manifest with action definition and controller reference
+    Object.assign(
+        getTargetElementReference(manifest, config.target),
+        JSON.parse(render(fs.read(join(root, `manifest.action.json`)), config))
+    );
+    fs.writeJSON(manifestPath, manifest);
+
+    // add event handler
+    fs.copyTpl(join(root, 'ext/EventHandler.js'), join(config.path, `${config.name}.js`), config);
```

Comment:
There should be an option to run generateCustomAction without this step. For scenarios that want to reuse the same event handler multiple times:
a) if you want to have the same extension action at different places of the UI
b) if you wat to use the same event handler for multiple actions (like in the FPM explorer example)

Reply:
Agree, also the coding for event handling of a custom column and a custom action should be the same. Will take care of it.

Reply:
c866f3d

Reply:
Great, thanks